core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -479,7 +479,10 @@ protected Node visitShowCreate(ShowCreate node, Void context)
 
                 accessControl.checkCanShowCreateTable(session.toSecurityContext(), new QualifiedObjectName(catalogName.getValue(), schemaName.getValue(), tableName.getValue()));
 
-                String sql = formatSql(new CreateView(QualifiedName.of(ImmutableList.of(catalogName, schemaName, tableName)), query, false, viewDefinition.get().getComment(), Optional.empty())).trim();
+                Optional<CreateView.Security> security = Optional.of(viewDefinition.get().isRunAsInvoker()
```

Comment:
nit : 
```suggestion
                CreateView.Security security = Optional.of(viewDefinition.get().isRunAsInvoker()
```

And pass it as Optional.of(security) in CreateView ?

Reply:
Good idea, thanks